### ISSUE TITLE
Unable to Display PDF: where url (folder or file name with space)

### ISSUE THREAD

**Issue by fredfgithud** [stephanrauh/ngx-extended-pdf-viewer-issues#13]
ex. 'https://childrensbooksforever.com/Childrenpics/A%20COOL%20KID%20LIKE%20ME.pdf'

**Comment by stephanrauh** [comment]
Mind to fill in some details? I've just tried https://pdfviewer.net/smartphone, and I can see the "cool kid" story without problems, despite the spaces in the file name.

**Comment by fredfgithud** [comment]
Hi Stephan Rauh,

Thank you for helping,

Which is file need to set to true?  ignoreResponsiveCSS

I am using angular 8, VS2019

I admire your App tool

I am new to angular

Thanks again

Fred
________________________________

**Comment by stephanrauh** [comment]
Thanks for the nice words! However, I don't understand your explanation. What does `ignoreResponsiveCSS` have to do with not being able to display a filename with blanks?

**Comment by fredfgithud** [comment]
Hi Stephan,
 I am using desktop, Please see blow,  Did I miss sometime or do it wrong


1.0

    <ngx-extended-pdf-viewer [(src)]="pdfViewerSrc"

                             [useBrowserLocale]="true"

                             [textLayer]="true"

                             [showHandToolButton]="true"

                             [height]="'85vh'">

    </ngx-extended-pdf-viewer>



1.1

[cid:50b1ab38-15b8-4a65-b4c4-c141b348a95b]

1.2

[cid:fb32b2e1-af82-4e7f-88c5-cd19ed3ac4ee]


1.3

[cid:e7fcb67d-b7d6-48f7-91f8-ca141729f00e]



2. Download and display

[cid:6de506df-4c81-4fcc-8331-e1add2500cee]


Thank you,

Fred



________________________________

**Comment by stephanrauh** [comment]
Oops - that's the first time that happened, but when I added the syntax highlighting to your post, the `[cid:xxxx]` items occured. Did I destroy your screenshots?

**Comment by fredfgithud** [comment]
1.0    <ngx-extended-pdf-viewer [(src)]="pdfViewerSrc"  [useBrowserLocale]="true"

                        [textLayer]="true” [showHandToolButton]="true" [height]="'85vh'">

         </ngx-extended-pdf-viewer>



1.1[cid:a07e9bbf-f334-4a4a-943f-0b4c18eef112]

1.2[cid:098bda49-459b-4121-89d6-58e9fc68d199]


1.3[cid:d5e294d6-4f57-4d1a-8e07-9c8361802a08]



2. Download and display[cid:856d6f8a-308f-49c1-b4e9-129aca5b31ac]






________________________________

**Comment by stephanrauh** [comment]
Hmm... maybe it wasn't my fault. Your latest addition shows like this on my PC:

![image](https://user-images.githubusercontent.com/3045767/85793311-1e6f9400-b735-11ea-9c5d-ff4c96fec49b.png)

I suppose that's not what you intended to send me? What does it look like on your side?

**Comment by fredfgithud** [comment]
Hi Stephan,

Sorry about that, You can tell I am beginner,
Thanks for your time.

Fred


________________________________

**Comment by stephanrauh** [comment]
Never mind. I'll help you if I can. I just have to know what the bug is. :)

**Comment by fredfgithud** [comment]
Hi Stephan,

Thanks in advance,

component.html

<div class="Container">
  <div class="form-group">
    <label>File input: </label>
    <input (change)="onFileSelected()" type="file" id="file">
    <label>Page #: </label>
    <input [(ngModel)]="page" type="text" id="page">
  </div>
  <div>
    <ngx-extended-pdf-viewer [src]="pdfNgxvSrc"
                              [useBrowserLocale]="true"
                              [textLayer]="true"
                              [showHandToolButton]="true"
                              [height]="'90vh'">
    </ngx-extended-pdf-viewer>
  </div>
</div>


******
component.ts

import { Component, OnInit } from '@angular/core';

@Component({
  selector: 'app-pdf-ngxextended',
  templateUrl: './pdf-ngxextended.component.html',
  styleUrls: ['./pdf-ngxextended.component.css']
})
export class PdfNgxextendedComponent implements OnInit {

  page: number = 1;
  pdfNgxvSrc: string = '';

  //Testing Links
  //filePath: string = 'https://vadimdez.github.io/ng2-pdf-viewer/assets/pdf-test.pdf';
  //filePath: string = 'https://childrensbooksforever.com/Childrenpics/A%20COOL%20KID%20LIKE%20ME.pdf';
  filePath: string = 'http://childrensbooksforever.com/Childrenpics/A%20COOL%20KID%20LIKE%20ME.pdf';

  constructor() { }

  ngOnInit() {
    //this.pdfNgxvSrc = this.pdfxService.getPDF();
    //this.pdfNgxvSrc = this.filePath.replace('%20', ' ').replace('%20', ' ').replace('%20', ' ').replace('%20', ' ') ;
    this.pdfNgxvSrc = this.filePath;
  }

  // event for onchange()
  onFileSelected() {

    // const inputTag = document.getElementById('file') as HTMLInputElement;
    //  const value = inputTag.value;

    let img: any = document.querySelector("#file");
    if (typeof (FileReader) !== 'undefined') {
      let reader = new FileReader();

      reader.onload = (e: any) => {
        this.pdfNgxvSrc = e.target.result;
      }
      reader.readAsArrayBuffer(img.files[0]);
    }
  }

}


input file:
http://childrensbooksforever.com/Childrenpics/A COOL KID LIKE ME.pdf
http://childrensbooksforever.com/Childrenpics/A%20COOL%20KID%20LIKE%20ME.pdf
https://childrensbooksforever.com/Childrenpics/A COOL KID LIKE ME.pdf
https://childrensbooksforever.com/Childrenpics/A%20COOL%20KID%20LIKE%20ME.pdf<http://childrensbooksforever.com/Childrenpics/A%20COOL%20KID%20LIKE%20ME.pdf>


Thanks again,

Fred





________________________________

**Comment by fredfgithud** [comment]
Hi Stephan,

how to assign the link to below able view to the pdf?

A COOL KID LIKE ME.pdf
https://childrensbooksforever.com/Childrenpics/A COOL KID LIKE ME.pdf<http://childrensbooksforever.com/Childrenpics/A%20COOL%20KID%20LIKE%20ME.pdf>


Fred


________________________________

**Comment by stephanrauh** [comment]
Are you aware of the CORS problem? Maybe that's the cause of your trouble.

The showcase doesn't have problems with the spaces in the filename: https://github.com/stephanrauh/extended-pdf-viewer-showcase/blob/ef84fa1e38b299b2017850c0f9b4af37f93787aa/src/app/smartphone/smartphone.component.html#L31

**Comment by stephanrauh** [comment]
CORS is a protection against hackers. As a side effect, it also drives developers nuts.

Have a look at https://juristr.com/blog/2016/11/configure-proxy-api-angular-cli/ for in-depth information and a solution. However, I'm not sure if Juri's `proxy.conf.json` file still works. When I needed it two days ago, I had to use a slightly different version:

```json
{
  "/api": {
    "target": {
      "host": "www.beyondjava.net",
      "protocol": "https:",
      "port": 443
    },
    "secure": true,
    "logLevel": "debug",
    "changeOrigin": true
  }
}
```

**Comment by fredfgithud** [comment]
Hi Stephan,

Thank you for quick response,

Yes, seem like it CORS issue
I was try to config in backed code, seem like configured wrong,

I will try your Proxy.config.json

I see you new and enhancement deployment

Congratulation,

Thanks again,

Fred

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows 10

**Comment by stephanrauh** [comment]
Awesome! Alway glad to help!

Best regards,
Stephan
